tools/walletextension/README.md

```diff
@@ -0,0 +1,50 @@
+# Wallet extension
+
+The wallet extension is an HTTP server that sits between the Obscuro client and the Obscuro host. It encrypts outgoing 
+requests with the enclave public key, and decrypts any encrypted responses from the enclave (i.e. responses to 
+`eth_getBalance` and `eth_getStorageAt` requests) with a _viewing key_.
+
+A _viewing key_ is a public key generated by the wallet extension. The public key is sent to the enclave, accompanied 
+by a signature proving the public key was generated by a specific account. The private key is stored locally by the 
+wallet extension. This viewing key and its corresponding private key are then used to securely encrypt messages from 
+the enclave to the wallet extension.
+
+The signature over the viewing key is created using [MetaMask](https://metamask.io/). Currently, other signers are not 
+supported.
+
+The wallet extension should be run locally, such that no sensitive data leaves the client's machine unencrypted.
+
+## Obscuro facade
+
+The Obscuro facade is an additional tool that sits between the wallet extension and a regular Ethereum node. It 
+decrypts requests using the enclave public key, and encrypts sensitive responses using the viewing key. In this way, it 
```

Comment:
using the viewing key associated with the "from" of the eth_call, or the eth_balance